packages/fx-core/src/component/driver/deploy/azure/AzureZipDeployDriver.ts

```diff
@@ -0,0 +1,212 @@
+
+export class AzureZipDeployDriver extends AzureDeployDriver {
+  pattern =
+    /\/subscriptions\/([^\/]*)\/resourceGroups\/([^\/]*)\/providers\/Microsoft.Web\/sites\/([^\/]*)/i;
+  private readonly serviceName: string;
+  protected helpLink = "https://aka.ms/teamsfx-actions/azure-app-service-deploy";
+  protected summaries: string[];
+  protected summaryPrepare: string[];
+  protected zipBuffer: Buffer | undefined;
+
+  constructor(
+    args: unknown,
+    context: DriverContext,
+    serviceName: string,
+    summaries: string[],
+    summaryPrepare: string[]
+  ) {
+    super(args, context);
+    this.serviceName = serviceName;
+    this.summaries = summaries;
+    this.summaryPrepare = summaryPrepare;
+  }
+
+  async azureDeploy(
+    args: DeployStepArgs,
+    azureResource: AzureResourceInfo,
+    azureCredential: TokenCredential
+  ): Promise<void> {
+    await this.progressBar?.start();
+    const cost = await this.zipDeploy(args, azureResource, azureCredential);
+    await this.progressBar?.next(ProgressMessages.restartAzureService);
+    await this.restartFunctionApp(azureResource);
+    await this.progressBar?.end(true);
+    if (cost > DeployConstant.DEPLOY_OVER_TIME) {
+      await this.context.logProvider?.info(
+        getLocalizedMessage(
+          "driver.deploy.notice.deployAcceleration",
+          "https://aka.ms/teamsfx-config-run-from-package"
+        ).localized
+      );
+    }
+  }
+
+  protected prepare: (args: DeployStepArgs) => Promise<void> = async (args: DeployStepArgs) => {
+    await this.progressBar?.next(ProgressMessages.packingCode);
+    await this.packageToZip(args, this.context);
+  };
+
+  /**
+   * deploy to azure app service or azure function use zip deploy method
+   * @param args local file needed to be deployed
+   * @param azureResource azure resource info
+   * @param azureCredential azure user login credential
+   * @return the zip deploy time cost
+   * @protected
+   */
+  public async zipDeploy(
+    args: DeployStepArgs,
+    azureResource: AzureResourceInfo,
+    azureCredential: TokenCredential
+  ): Promise<number> {
+    await this.progressBar?.next(ProgressMessages.packingCode);
+    const zipBuffer = await this.packageToZip(args, this.context);
+    await this.progressBar?.next(ProgressMessages.getAzureAccountInfoForDeploy);
+    await this.context.logProvider.debug("Start to get Azure account info for deploy");
+    const config = await this.createAzureDeployConfig(azureResource, azureCredential);
+    await this.context.logProvider.debug("Get Azure account info for deploy complete");
+    await this.progressBar?.next(ProgressMessages.getAzureUploadEndpoint);
+    const endpoint = this.getZipDeployEndpoint(azureResource.instanceId);
+    await this.context.logProvider.debug(`Start to upload code to ${endpoint}`);
+    await this.progressBar?.next(ProgressMessages.uploadZipFileToAzure);
+    const startTime = Date.now();
```

Comment:
Nit: One of my suggestions is to use existing lib, for example, some Timer lib, for tracking time spent.

Reply:
I can't find Timer lib usage in our codes. May it have another name?